tests/pkg_tinyvcdiff/main.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2022 Juergen Fitschen
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       tinyvcdiff library backend tests
+ *
+ * @author      Juergen Fitschen <me@jue.yt>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include <fcntl.h>
+#include "board.h"
+#include "embUnit.h"
+#include "vcdiff.h"
+#include "vcdiff_mtd.h"
+#include "vcdiff_vfs.h"
+#include "mtd_mapper.h"
+#include "fs/fatfs.h"
+
+static const char *bin_source = "Hello world! I hope you are doing well ...\n";
+static const char *bin_target = "Hello universe! I hope you are doing well ...\n";
+static const uint8_t diff[] = {
+    0xd6, 0xc3, 0xc4, 0x53, 0x00, 0x01, 0x2b, 0x00, 0x17, 0x2e, 0x00, 0x00,
+    0x12, 0x00, 0x0f, 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x75, 0x6e, 0x69,
+    0x76, 0x65, 0x72, 0x73, 0x65, 0x13, 0x20, 0x0b
+};
+
+static void test_tinyvcdiff(void)
+{
+    /* Prepare MTD device */
+    mtd_power(MTD_0, MTD_POWER_UP);
+    TEST_ASSERT_EQUAL_INT(mtd_init(MTD_0), 0);
+    mtd_mapper_parent_t mtd_parent = MTD_PARENT_INIT(MTD_0);
+
+    /* 1. setup vcdiff */
+    vcdiff_t vcdiff;
+    vcdiff_init(&vcdiff);
+    vcdiff_set_logger(&vcdiff, printf, NULL);
+
+    /* 2. prepare source binary */
+    mtd_mapper_region_t mtd_source = {
+        .mtd = {
+            .driver = &mtd_mapper_driver,
+            .sector_count = mtd_parent.mtd->sector_count / 2,
+            .pages_per_sector = mtd_parent.mtd->pages_per_sector,
+            .page_size = mtd_parent.mtd->page_size
+        },
+        .parent = &mtd_parent,
+        .sector = 0
+    };
+    TEST_ASSERT_EQUAL_INT(0, mtd_write(&mtd_source.mtd, (uint8_t*)bin_source, 0, strlen(bin_source)));
+    vcdiff_mtd_t vcdiff_source = VCDIFF_MTD_INIT(&mtd_source.mtd);
+    vcdiff_set_source_driver(&vcdiff, &vcdiff_mtd_driver, (void*)&vcdiff_source);
```

Comment:
```suggestion
    vcdiff_set_source_driver(&vcdiff, &vcdiff_mtd_driver, &vcdiff_source);
```
should work too since `vcdiff_set_source_driver` takes a `void *`